test/e2e/storage/testsuites/volume_modify.go

```diff
@@ -0,0 +1,310 @@
+	e2evolume "k8s.io/kubernetes/test/e2e/framework/volume"
+	storageframework "k8s.io/kubernetes/test/e2e/storage/framework"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+const (
+	modifyPollInterval               = 2 * time.Second
+	setVACWaitPeriod                 = 30 * time.Second
+	modifyingConditionSyncWaitPeriod = 2 * time.Minute
+	modifyVolumeWaitPeriod           = 10 * time.Minute
+	vacCleanupWaitPeriod             = 30 * time.Second
+)
+
+type volumeModifyTestSuite struct {
+	tsInfo storageframework.TestSuiteInfo
+}
+
+// InitCustomVolumeModifyTestSuite returns volumeModifyTestSuite that implements TestSuite interface
+// using custom test patterns
+func InitCustomVolumeModifyTestSuite(patterns []storageframework.TestPattern) storageframework.TestSuite {
+	return &volumeModifyTestSuite{
+		tsInfo: storageframework.TestSuiteInfo{
+			Name:         "volume-modify",
+			TestPatterns: patterns,
+			SupportedSizeRange: e2evolume.SizeRange{
+				Min: "1Gi",
+			},
+			TestTags: []interface{}{feature.VolumeAttributesClass},
+		},
+	}
+}
+
+// InitVolumeModifyTestSuite returns volumeModifyTestSuite that implements TestSuite interface
+// using testsuite default patterns
+func InitVolumeModifyTestSuite() storageframework.TestSuite {
+	patterns := []storageframework.TestPattern{
+		storageframework.DefaultFsDynamicPV,
+		storageframework.BlockVolModeDynamicPV,
+		storageframework.NtfsDynamicPV,
+	}
+	return InitCustomVolumeModifyTestSuite(patterns)
+}
+
+func (v *volumeModifyTestSuite) GetTestSuiteInfo() storageframework.TestSuiteInfo {
+	return v.tsInfo
+}
+
+func (v *volumeModifyTestSuite) SkipUnsupportedTests(driver storageframework.TestDriver, pattern storageframework.TestPattern) {
+	_, ok := driver.(storageframework.VolumeAttributesClassTestDriver)
+	if !ok {
+		e2eskipper.Skipf("Driver %q does not support VolumeAttributesClass tests - skipping", driver.GetDriverInfo().Name)
+	}
+	if !driver.GetDriverInfo().Capabilities[storageframework.CapBlock] && pattern.VolMode == v1.PersistentVolumeBlock {
+		e2eskipper.Skipf("Driver %q does not support block volume mode - skipping", driver.GetDriverInfo().Name)
+	}
+}
+
+func (v *volumeModifyTestSuite) DefineTests(driver storageframework.TestDriver, pattern storageframework.TestPattern) {
+	type local struct {
+		config *storageframework.PerTestConfig
+
+		resource *storageframework.VolumeResource
+		vac      *storagev1alpha1.VolumeAttributesClass
+	}
+	var l local
+
+	// Beware that it also registers an AfterEach which renders f unusable. Any code using
+	// f must run inside an It or Context callback.
+	f := framework.NewFrameworkWithCustomTimeouts("volume-modify", storageframework.GetDriverTimeouts(driver))
+	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
+
+	init := func(ctx context.Context, createVolumeWithVAC bool) {
+		l = local{}
+
+		// TODO for reviewers - are the group/version/resource stored as a constant anywhere I can access? should I just make my own constants?
```

Comment:
We don't need to add this check. It requires that the API and feature-gate enabled when a job in the test-infra repo is designed to run VolumeAttributesClass e2e tests.

The sig-test team are working on [RFC: e2e: enhance WithFeatureGate labels](https://github.com/kubernetes/kubernetes/pull/124350). Once it is completed, we only need to use WithFeatureGate in our tests, the kind-alpha/beta jobs owned by sig-test can test the VolumeAttributesClass feature.

But now, we still need to use both WithFeatureGate and WithFeature. the pull-kubernetes-e2e-storage-kind-alpha-features job is only one to run VolumeAttributesClass e2e tests.